vbox_create.sh

```diff
@@ -60,16 +56,23 @@ VBOX_DIR="`dirname ${BASH_SOURCE[0]}`/vbox"
 [[ -d $VBOX_DIR ]] || mkdir $VBOX_DIR
 P=`python -c "import os.path; print os.path.abspath(\"${VBOX_DIR}/\")"`
 
-# ensure cluster.txt has CLUSTER_PREFIX prepended before VM names
-# NOTE: somewhat crude we see if any line does not start with the
-# cluster name and if so all lines have it prepended
-if [[ -n "$(grep -v "^${CLUSTER_PREFIX}" ./cluster.txt)" ]]; then
-  sed -i "s/^/${CLUSTER_PREFIX}-/" cluster.txt
+# populate the VM_LIST array from cluster.txt
+# HACK: This is called prior to cluster.txt modification
+# we need to "manually" prepend BACH_CLUSTER_PREFIX to
+# all hostnames as we will generate VMs before cluster.txt
+# is edited
+if [ ${BACH_CLUSTER_PREFIX} == '']; then
```

Comment:
Could this whole block be replaced with a use of `ClusterData`? I'm thinking:
```code_to_produce_vm_list="
require './lib/cluster_data.rb';
include BACH::ClusterData;
cp=ENV.fetch('CLUSTER_PREFIX', '');
cp += '-' unless cp.empty?;
vms = parse_cluster_txt(File.readlines(File.join(
  'stub-environment','${CLUSTER_TYPE,,}_cluster.txt')))
puts vms.map{|e| cp + e[:hostname]}.join(' ')
"
VM_LIST=( $(ruby -e "$code_to_produce_vm_list") )
```

Particularly, we should read the `stub-environment` cluster.txt for the type we are building too as the symlink should be dangling at this point if we do not already have a `cluster` directory. (Don't we only create a concrete cluster.txt when we run [`vm-to-cluster.sh`](https://github.com/bloomberg/chef-bach/blob/29663e9e2b993bcc700f8cdffa60563422c9da04/vbox_create.sh#L264) below?)

Reply:
copying cluster.txt is one of the very first things we do 
https://github.com/bloomberg/chef-bach/blob/f1fd81a35e7383137b4734c198ed893a5b9f40ca/tests/automated_install.sh#L46-L56

Reply:
Ah, yes, indeed the code should use `cluster.txt` instead of `File.readlines(File.join(
  'stub-environment','${CLUSTER_TYPE,,}_cluster.txt'))`; I look forward to your final version. For what it is worth this tested well on a cluster build last night (with this merged on top of #1010) for me with two other clusters already installed.